cmd/immuadmin/command/hot_backup.go

```diff
@@ -0,0 +1,661 @@
+		}
+		err = cl.backupTx(i, output)
+		if err != nil {
+			return err
+		}
+		if bar != nil {
+			bar.Add(1)
+		}
+	}
+
+	fmt.Fprintf(os.Stderr, "Done\n")
+	return nil
+}
+
+func (cl *commandlineHotBck) backupTx(tx uint64, output io.Writer) error {
+	stream, err := cl.immuClient.ExportTx(cl.context, &schema.TxRequest{Tx: tx})
+	if err != nil {
+		return fmt.Errorf("failed to export transaction: %w", err)
+	}
+
+	var content []byte
+	for {
+		var chunk *schema.Chunk
+		chunk, err = stream.Recv()
+		if errors.Is(err, io.EOF) {
+			err = nil
+			break
+		} else if err != nil {
+			break
+		}
+		content = append(content, chunk.Content...)
+	}
+
+	if err != nil {
+		return fmt.Errorf("cannot process transaction data: %w", err)
+	}
+	err = stream.CloseSend()
+	if err != nil {
+		return fmt.Errorf("CloseSend returned %v", err)
+	}
+	txn, err := cl.immuClient.TxByID(cl.context, tx)
+	if err != nil {
+		return err
+	}
+	err = outputTx(tx, output, txn.Header.EH, content)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func outputTx(tx uint64, output io.Writer, signature []byte, content []byte) error {
+	payload := make([]byte, headerSize, headerSize+len(signature)+len(content))
+	copy(payload[prefixOffset:], prefix)
+	binary.BigEndian.PutUint32(payload[versionOffset:], latestFileVersion)
+	binary.BigEndian.PutUint64(payload[txIdOffset:], tx)
+	binary.BigEndian.PutUint32(payload[txSignSizeOffset:], uint32(len(signature)))
+	binary.BigEndian.PutUint32(payload[txSizeOffset:], uint32(len(content)))
+	payload = append(payload, signature...)
+	payload = append(payload, content...)
+
+	_, err := output.Write(payload)
+	return err
+}
+
+type restoreParams struct {
+	input    string
+	append   bool
+	progress bool
+	force    bool
+	verify   bool
+}
+
+func (cl *commandlineHotBck) hotRestore(cmd *cobra.Command) {
+	ccmd := &cobra.Command{
+		Use:   "hot-restore <db_name>",
+		Short: "Restore data from backup file",
+		Long: "Restore saved transaction from backup file without stopping the database engine. " +
+			"Restore can restore the data from scratch or apply only the missing data.",
+		PersistentPreRunE: cl.ConfigChain(cl.connect),
+		PersistentPostRun: cl.disconnect,
+		RunE: func(cmd *cobra.Command, args []string) error {
+
+			params, err := prepareRestoreParams(cmd.Flags())
+			if err != nil {
+				return err
+			}
+
+			file := io.Reader(os.Stdin)
+			if params.input != "-" {
+				f, err := os.Open(params.input)
+				if err != nil {
+					return err
+				}
+				file = f
+				defer f.Close()
+			}
+
+			if params.verify {
+				return verifyFile(file)
+			}
+
+			dbExist, err := cl.isDbExists(args[0])
+			if err != nil {
+				return err
+			}
+
+			var signature, payload []byte
+			var firstTx uint64
+			if dbExist {
+				firstTx, signature, payload, err = cl.verifyDb(params, args[0], file)
+				if err != nil {
+					return err
+				}
+				err = cl.immuClient.UpdateDatabase(cl.context, &schema.DatabaseSettings{DatabaseName: args[0], Replica: true})
```

Comment:
automatic switching replica mode on/off might lead to some issues in the future, because UpdateDatabase may expect more settings which may get lost as part of this process...

Probably it's safe to switch it off upon completion in the database is created in this restore process

Reply:
Can you please elaborate on this? How it should work?

Reply:
if restore is called into existent database it will fail if it's not in replica mode.
if restore is called into non-existent database, it could be automatically created as replica and may be set as non-replica when all the transactions has been applied.

I'd discourage calling update if there is not knowledge about all the settings of the database.

Reply:
So you are suggesting that user need to manually switch existing database to replica mode before running restore? This isn't very user friendly. Maybe it is better to add CLI option to force switching to `replica`?

Reply:
good idea about the flag, it actually could be the "force" flag...

Reply:
Added `force-replica` flag